annif/util.py

```diff
@@ -13,21 +14,31 @@ def atomic_save(obj, dirname, filename, method=None):
     """Save the given object (which must have a .save() method, unless the
     method parameter is given) into the given directory with the given
     filename, using a temporary file and renaming the temporary file to the
-    final name."""
-
-    prefix, suffix = os.path.splitext(filename)
-    tempfd, tempfilename = tempfile.mkstemp(
-        prefix=prefix, suffix=suffix, dir=dirname)
-    os.close(tempfd)
+    final name. To save a directory explicitly set filename=None."""
+
+    if filename:
```

Comment:
There isn't that much shared code between the file vs. directory cases. I wonder if it would be better to leave the `atomic_save` function as it is (only supporting files, not directories) and adding a new function `atomic_save_dir` which takes care of directories only. The implementations could be simpler, without so many if clauses.